src/api/app/services/user/admin_rights.rb

```diff
@@ -0,0 +1,23 @@
+class User::AdminRights
+  attr_reader :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def toggle!
+    raise NotFoundError unless user
+
+    if user.roles.exists?(title: 'Admin')
+      RolesUser.where(user: user, role: admin_role).first.destroy
+    else
+      user.roles << admin_role
+    end
+
+    user
+  end
+
+  def admin_role
+    @admin_role ||= Role.global.where(title: 'Admin').first
+  end
+end
```

Comment:
This really should not be part of the app, we don't have a need for a service doing this. Especially not without any authorization.

Reply:
Ok, I'll move it to `lib/`

Reply:
I've moved the code to `lib/'. Check it out again, please!